Annotator method with documentation

PrimozGodec opened the pull request.

gokceneraslan committed:
Fix violinplot not being produced

When groupby is not None and multipanel is True, nothing is plotted. Now we fallback to non-multipanel if groupby is specified.

gokceneraslan committed:
Fix the last multipanel condition

flying-sheep committed:
Fix many many documentation issues

flying-sheep committed:
Suppress FutureWarning when building docs

PrimozGodec committed:
Annotator method with documentation

flying-sheep committed:
Automatically document settings

flying-sheep committed:
Ignore scanpy.api for broken links, fix some links

flying-sheep committed:
Parameter type fixes

flying-sheep committed:
More parameter type fixes

flying-sheep committed:
Fix combat type inference

flying-sheep committed:
Whoops, fixed obsm/varm test

gamazeps committed:
Pass on copy and return_info to dca (#791)

The arguments were not passed in preprocessing/_dca.py, which
makes debugging much more complicated  than needed.

flying-sheep committed:
More param type fixes

flying-sheep committed:
MAGIC param type fixes

flying-sheep committed:
PHATE param type fixes

flying-sheep committed:
Remaining param type fixes

flying-sheep committed:
typo

flying-sheep committed:
Don’t let broken docs compile

flying-sheep committed:
Warn about parameters types that aren’t up to speed.

flying-sheep committed:
Elaborate on function docs

flying-sheep committed:
Remove broken RNA velocity code and reference scvelo (#795)

flying-sheep committed:
Migrate to setuptools_scm and get rid of packaging cruft. (#798)

No vendored versioneer.py anymore, no setup.cfg!

flying-sheep committed:
We also don’t need docs in the sdist

fidelram committed:
Embedding improvements (#794)

- Provide add_contour option to scatterplots (embeddings)
- Add `sc.pl.embedding` to the API which can plot any basis (#762). Added vmin and vmax  (#775)

ivirshup committed:
Add setuptools_scm to docs/requirements.txt

ivirshup committed:
Remove erroneous reference from scanpy/api/__init__.py

ivirshup committed:
Fix docstring typo

mstrazar commented on scanpy/external/_tools/_annotator.py:
```diff
@@ -0,0 +1,116 @@
+    ...     "0.1.0/DC_expMatrix_DCnMono.csv.gz")
+    >>> file_name_markers, _ = urllib.request.urlretrieve(
+    ...     "https://github.com/PrimozGodec/cell-annotation/releases/download/"
+    ...     "0.1.0/panglao_gene_markers.csv.gz")
```
The main criterion for this function to be used by the community will be simplicity. The current marker table `panglao_gene_markers.csv.gz"` can be simplified to  skip some steps below.

A link to the Panglao database needs to be added somewhere, ideally in the function documentation along with citation.

PrimozGodec replied:
It is simplified and citation is added.

mstrazar commented on scanpy/external/_tools/_annotator.py:
```diff
@@ -0,0 +1,116 @@
+        Tabular data with gene expressions.
+    markers : pd.DataFrame
+        The data-frame with marker genes and cell types. Data-frame has two
+        columns Gene and Cell Type first holds gene names or ID and second cell
```
Emphasize Gene and Cell Type names.

PrimozGodec replied:
Done

mstrazar commented on scanpy/external/_tools/_annotator.py:
```diff
@@ -0,0 +1,116 @@
+    >>> df = pd.read_csv(file_name_data, compression="gzip").iloc[:, :-2]
+    >>> df_markers = pd.read_csv(file_name_markers, compression="gzip")
+    >>> # current markers have wrong name for Gene column
+    >>> df_markers = df_markers.rename(columns={'Name': 'Gene'})
```
Can you prepare the data such that this renaming is not necessary?

PrimozGodec replied:
Done

mstrazar commented on scanpy/external/_tools/_annotator.py:
```diff
@@ -0,0 +1,116 @@
+    >>> # current markers have wrong name for Gene column
+    >>> df_markers = df_markers.rename(columns={'Name': 'Gene'})
+    >>> # we need just marker genes for human
+    >>> df_markers = df_markers[df_markers["Organism"] == "Human"]
```
Prepare the data such that the markers are already human only.

PrimozGodec replied:
Done

mstrazar commented on scanpy/external/_tools/_annotator.py:
```diff
@@ -0,0 +1,116 @@
+    >>> df_markers = df_markers[df_markers["Organism"] == "Human"]
+    >>>
+    >>> # transform data to AnnData
+    >>> anndata = AnnData(df.values, var=df.columns.values)
```
Can `AnnData` be constructed directly from a file, in one line?
https://scanpy.readthedocs.io/en/stable/basic_usage.html#anndata

PrimozGodec replied:
It can be constructed with `read_csv` function but this function cannot uncompress the data what pandas can. Without compression, data wold be 130MB instead of 18MB now. Maybe we can replace data with AML for this example?

mstrazar commented on scanpy/external/_tools/_annotator.py:
```diff
@@ -0,0 +1,116 @@
+        columns Gene and Cell Type first holds gene names or ID and second cell
+        type for this gene. Gene names must be written in the same format than
+        genes in `adata`.
+    num_genes : int
```
The current way is a bit vague because you have to know this value.

If no value is provided, use the number of columns in the gene expression matrix. Perhaps issue a warning or message when this happens.

PrimozGodec replied:
Done

fidelram committed:
fix vmin/vmax for categorical data #800

fidelram committed:
Merge pull request #804 from fidelram/fix_800_vmin

fix vmin/vmax for categorical data #800

Commit:
Add test.

fidelram committed:
Merge pull request #776 from theislab/violin-multipanel-fix

Fix sc.pl.violin not making violinplots

gokceneraslan committed:
Fix typo in function name _plot_color_cycle

gokceneraslan committed:
Add fontoutline shortcut to paga plotting functions and remove empty dicts as default arguments.

gokceneraslan committed:
Workaround for 'panel_pos local variable referenced before assignment.' error

gokceneraslan committed:
Fix missing spaces in log messages in sc.pp.filter_genes and filter_cells

gokceneraslan committed:
Exit with an error if sc.pl.pca_loadings is called with indices < 1 (#803)

ivirshup committed:
Allow specifying layer in rank_genes_groups (#801)

flying-sheep committed:
Improve cellbrowser docs and error handling

flying-sheep committed:
Vastly simplified palantir code

flying-sheep committed:
Style: trailing commas

flying-sheep committed:
Simplified describe_obs

Commit:
Add PAGA plotting tests

gokceneraslan committed:
Show PCA loadings for genes with lowest loadings (#805)

* Add include_lowest option to pca_loadings which allows us to plot genes with negative loadings

* Hide xticks if include_lowest is given

* Add test.

* Fix string components and extend the test

* Fix pca_loadsings test after making include_lowest True by default

flying-sheep committed:
Adapt coding style guidelines

flying-sheep committed:
Style

flying-sheep committed:
ranking style fix

gamazeps committed:
Make the dca learning rate accessible from scanpy. (#813)

flying-sheep committed:
scanpydoc 0.4

flying-sheep committed:
Fixed style and docs for datasets

flying-sheep committed:
Moved external tools to scanpy.external.tl (#814)

flying-sheep committed:
Moved things belonging to external.pp into there

flying-sheep committed:
Completed docs for external module, rm’d defunct rtools, moved external

gokceneraslan committed:
Fix clustermap error due to fillna call in seaborn. (#809)

gokceneraslan committed:
Add obs_key support to PAGA plots and fix the cax parameter.

gokceneraslan committed:
Do not overwrite cax variable and add another paga plotting test.

flying-sheep committed:
Only make broken links cause a doc build failure when on readthedocs

Errors cause cache invalidation, making sphinx very slow on repeated builds.
We want the docs to fail building on rtd (to see errors) but locally warnings are sufficient.

flying-sheep committed:
Fix performance regression caused by

flying-sheep committed:
Simplify scatter and its docs

flying-sheep committed:
Remove defunct _top_genes_visual.py

flying-sheep committed:
Harmonized docs and params of embedding_density and the embedding plots

flying-sheep committed:
Reenable docs tests

flying-sheep committed:
Better test IDs

flying-sheep committed:
Renamed all used utils to _*

flying-sheep committed:
Adopt changes from PR (Fixes #656)

flying-sheep committed:
Adapt to anndata style

flying-sheep committed:
Link to stable AnnData

flying-sheep committed:
Fix weird discrepancy between travis and local

flying-sheep committed:
Fix matplotlib links

ivirshup committed:
Don't allow h5py 2.10.0

flying-sheep committed:
Added verbosity override

gokceneraslan committed:
Rotate only xticks in sc.pl.violin (#825)

Commit:
Fix fontoutline variable overwrite and add tests.

Commit:
Add forgotten test image

LuckyMD committed:
highly_variable_gene batch_key variant fix (#824)

flying-sheep committed:
sphinx-autodoc-typehints is fixed

flying-sheep committed:
Added CSS colors

flying-sheep committed:
Unify all isinstance calls (#839)

flying-sheep committed:
Don’t error if n_top_genes > n_var (#835)

* Don’t error if n_top_genes > n_var

* Whoops

flying-sheep committed:
blacken everything without too many line changes (#841)

ivirshup committed:
Use cacheing and parallelism for qc metrics

ivirshup committed:
Add release note for queries

Additionally some module doc updates.

ivirshup committed:
Run static analysis concurrently with tests (#846)

flying-sheep committed:
added cache_compression kwarg to read() (#831)

flying-sheep committed:
Added h5py intersphinx link

ivirshup committed:
Fix h5py doc link

h5py docs don't do https.

Koncopd committed:
fix zheng17 and recipe_seurat (#816)

flying-sheep committed:
Add cache_compression setting (#847)

ivirshup committed:
Add warning for qc_metric parallel

flying-sheep committed:
Fix many plotting idiosyncrasies. Fixes #850

flying-sheep committed:
Cleaned up unused utils

ivirshup committed:
Add note about compilation time to qc metrics

flying-sheep committed:
Better conda badge

flying-sheep committed:
Fix badge

tomwhite committed:
Add ability to run nearest neighbors, umap and louvain on a GPU using RAPIDS (#830)

flying-sheep committed:
better link in readme

fbnrst committed:
update ValueError message in pca (#858)

replaced the deprecated filter_gene_dispersion with highly_variable_genes

PrimozGodec committed:
Added unittests for annotator

PrimozGodec committed:
Added unittests for annotator

flying-sheep committed:
Fix docs

PrimozGodec committed:
Merge

PrimozGodec committed:
References

PrimozGodec committed:
Fixes
